src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/ValidateRespondentWitnesses.java

```diff
@@ -0,0 +1,88 @@
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import static uk.gov.hmcts.reform.civil.callback.CallbackParams.Params.BEARER_TOKEN;
+import static uk.gov.hmcts.reform.civil.enums.CaseRole.RESPONDENTSOLICITORONE;
+import static uk.gov.hmcts.reform.civil.enums.CaseRole.RESPONDENTSOLICITORTWO;
+import static uk.gov.hmcts.reform.civil.enums.MultiPartyScenario.ONE_V_ONE;
+import static uk.gov.hmcts.reform.civil.enums.YesOrNo.NO;
+import static uk.gov.hmcts.reform.civil.enums.YesOrNo.YES;
+import static uk.gov.hmcts.reform.civil.service.flowstate.FlowFlag.TWO_RESPONDENT_REPRESENTATIVES;
+
+@Component
+public class ValidateRespondentWitnesses implements CaseTask, WitnessesValidator {
+
+    private final IStateFlowEngine stateFlowEngine;
+    private final CoreCaseUserService coreCaseUserService;
+    private final UserService userService;
+
+    public ValidateRespondentWitnesses(IStateFlowEngine stateFlowEngine, CoreCaseUserService coreCaseUserService, UserService userService) {
+        this.stateFlowEngine = stateFlowEngine;
+        this.coreCaseUserService = coreCaseUserService;
+        this.userService = userService;
+    }
+
+    @Override
+    public CallbackResponse execute(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+
+        if (ONE_V_ONE.equals(MultiPartyScenario.getMultiPartyScenario(caseData))) {
+            return validateWitnesses(caseData.getRespondent1DQ());
+        }
+
+        if (solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORONE)) {
+            return validateWitnesses(caseData.getRespondent1DQ());
+
+        }
+        if (solicitorRepresentsOnlyOneOfRespondents(callbackParams, RESPONDENTSOLICITORTWO)) {
+            return validateWitnesses(caseData.getRespondent2DQ());
+        }
+
+        if (hasRespondent2DifferentResponseAndWitnesses(caseData)) {
+            return validateWitnesses(callbackParams.getCaseData().getRespondent2DQ());
+        }
+
+        return validateWitnesses(caseData.getRespondent1DQ());
+    }
+
+    private boolean hasRespondent2DifferentResponseAndWitnesses(CaseData caseData) {
+        return respondent2HasSameLegalRep(caseData)
+            && caseData.getRespondentResponseIsSame() == NO
+            && caseData.getRespondent2DQ() != null
+            && caseData.getRespondent2DQ().getRespondent2DQWitnesses() != null;
+    }
+
+    private boolean solicitorRepresentsOnlyOneOfRespondents(CallbackParams callbackParams, CaseRole caseRole) {
+        return solicitorRepresentsOnlyOneOrBothRespondents(callbackParams, caseRole);
+    }
+
+    private boolean respondent2HasSameLegalRep(CaseData caseData) {
+        return caseData.getRespondent2SameLegalRepresentative() != null
+            && caseData.getRespondent2SameLegalRepresentative() == YES;
+    }
+
+    private boolean solicitorRepresentsOnlyOneOrBothRespondents(CallbackParams callbackParams, CaseRole caseRole) {
```

Comment:
first, isSomething
second, the method name could be better, the flag is giving the clue. why not calling it isTwoRespondentRepresentative?

Reply:
in the other comments you just mentionned that you wanted the is pattern isSolicitorRepresentingOneOrBothRespondents descriptive enough ?